docs/content/configuration.md

```diff
@@ -283,7 +283,53 @@ Following successful authentication at the token endpoint the returned token wil
 | `services[_].credentials.oauth2.client_secret` | `string` | Yes | The client secret to use for authentication. |
 | `services[_].credentials.oauth2.scopes` | `[]string` | No | Optional list of scopes to request for the token. |
 
-#### AWS signature
+#### OAuth2 JWT Bearer Grant Type
+
+OPA will authenticate using a bearer token obtained through the OAuth2 [JWT authorization grant](https://tools.ietf.org/html/rfc7523#section-2.1) flow.
+Rather than providing a client secret along with the request for an access token, the client [asserts](https://tools.ietf.org/html/rfc7521#section-4.1) its identity in the form of a signed JWT.
+
+| Field | Type | Required | Description |
+| --- | --- | --- | --- |
+| `services[_].credentials.oauth2.token_url` | `string` | Yes | URL pointing to the token endpoint at the OAuth2 authorization server. |
+| `services[_].credentials.oauth2.grant_type` | `string` | Yes | Must be set to `jwt_bearer`. |
+| `services[_].credentials.oauth2.signing_key` | `string` | Yes | Optional list of scopes to request for the token. |
+| `services[_].credentials.oauth2.additional_claims` | `map` | No | Map of claims to include in the JWT (see notes below) |
+| `services[_].credentials.oauth2.include_jti_claim` | `bool` | No | Include a uniquely generated `jti` claim in any issued JWT |
+| `services[_].credentials.oauth2.scopes` | `[]string` | No | Optional list of scopes to request for the token. |
+
+Two claims will always be included in the issued JWT: `iat` and `exp`. Any other claims will be populated from the `additional_claims` map.
+
+##### Example
+
+Using a [Google Cloud Storage](https://cloud.google.com/storage/) bucket as a bundle service backend from outside the
+cloud account (for access from inside the account, see the [GCP Metadata Token](#GCP Metadata Token) section).
+
+```yaml
+services:
+  gcp:
+    url: ${BUNDLE_SERVICE_URL}
+    credentials:
+      oauth2:
+        grant_type: jwt_bearer
+        signing_key: jwt_signing_key
```

Comment:
Can we have a comment here? Something like
```suggestion
        signing_key: jwt_signing_key # references the key in `keys` below
```
just to make it extra clear at a glance :)

Reply:
Yeah that's a good idea. Will add.